lib/internal/test_runner/reporter/tap.js

```diff
@@ -154,11 +159,16 @@ function jsToYaml(indent, name, value, seen) {
   seen.add(value);
   const entries = ObjectEntries(value);
   const isErrorObj = isError(value);
-  let result = '';
   let propsIndent = indent;
+  let result = '';
 
   if (name != null) {
-    result += `${indent}  ${name}:\n`;
+    result += prefix;
+    if (value instanceof Date) {
+      // YAML uses the ISO-8601 standard to express dates.
+      result += ' ' + value.toISOString();
+    }
```

Comment:
Using `instanceof Date` is problematic – for performance reasons (at least it was the case a few years ago) and for correctness-sake (e.g. `value instanceof Date` could return true, yet `value.toISOString` might not be a function). Instead you can get this information from `internalBindings('types').isDate`
```suggestion
    if (internalBindings('types').isDate(value)) {
      // YAML uses the ISO-8601 standard to express dates.
      result += ' ' + DatePrototypeToISOString(value);
    }
```

Reply:
@aduh95 suggestion seems good but then you have a dangling `Date` const destructure at the top of the file (from `primordials`).

Reply:
The solution seems simple: just remove the unused `Date` assignment. Or am I missing something?